packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts

```diff
@@ -200,4 +245,54 @@ export class HttpApi extends Resource implements IHttpApi {
       integration: options.integration,
     }));
   }
+
+  /**
+   * Add a custom domain for this API and create the API mapping to a stage.
+   */
+  public addDomainName(options: AddDomainNameOptions): DomainName {
+    const dn = new DomainName(this, `DomainName${options.domainName}`, {
+      domainName: options.domainName,
+      certificate: options.certificate,
+    });
+
+    // const stage = options.stage ?? this.defaultStage!;
+
+    // const mapping = new HttpApiMapping(this, `${dn.domainName}defaultMapping`, {
+    //   api: this,
+    //   domainName: dn,
+    //   stage,
+    // });
+    // mapping.node.addDependency(stage);
+
+    this._domainName = dn;
+
+    return dn;
+  }
+
+  /**
+   * Create the `HttpApiMapping` resource with the custom domain and stage for the HTTP API
+   * @param options Options for domain name mapping
+   */
+  public addDomainMapping(options: DomainMappingOptions) {
+    const mappingKey = options.mappingKey;
+    // const dn = this.addDomainName(options);
+    const stage = options.stage ?? this.defaultStage!;
+    // create the mapping
+    const mapping = new CfnApiMapping(this, `${options.domainName}${mappingKey}`, {
+      apiId: this.httpApiId,
+      domainName: options.domainName.domainName,
+      stage: stage.stageName,
+      apiMappingKey: mappingKey,
+    });
+    mapping.node.addDependency(options.domainName, stage);
+  }
+
+  /**
+   * The first domain name mapped to this API, if defined through the `domainName`
+   * configuration prop, or added via `addDomainName`
+   */
+  public get domainName() {
+    return this._domainName;
+  }
```

Comment:
Drop.

This is the wrong association. An `HttpApi` does not have a `domainName`.